### PR TITLE
fix: forward the sample rate from the message

### LINKED ISSUE #145: The sample rate on the event is not forwarded
**Versions**

Lambda runtime: al2023, arm64
Lambda extension: 11.2.3

**Steps to reproduce**

1. Set `SampleRate` to, say, 5 in an app
2. Deploy it
3. The events are properly sampled, but by the time they make it to honeycomb there is no sample rate, so the aggregates are wrong.

**Additional context**

I think the issue is in here https://github.com/honeycombio/honeycomb-lambda-extension/blob/3519812b63459ff19e7cb58b61f2677e7871377e/logsapi/server.go#L94

We're properly parsing the timestamp and the `data`, but we've neglected to copy the sample rate

### PR DESCRIPTION
**Pull request by NLincoln**
If the app that we're forwarding telemetry for is doing it's own
sampling, we want to honor that

Closes #145

### CODE REVIEW

File: logsapi/server.go [hunk 1]
```diff
@@ -111,6 +112,29 @@ func parseMessageTimestamp(event *libhoney.Event, msg LogMessage) time.Time {
...
+
+	if ok {
+		// duration_ms may be a float (e.g. 43.23), integer (e.g. 54) or a string (e.g. "43")
+		switch sampleRate := rate.(type) {
+		case float64:
+			foundRate = int(sampleRate)
+		case int64:
+			foundRate = int(sampleRate)
+		case string:
+			if d, err := strconv.ParseInt(sampleRate, 10, 32); err == nil {
```
Comment (kentquirk) [inline logsapi/server.go:127]: Since you're putting it into an int anyway, you could use Atoi.

**Review by kentquirk — APPROVED** [review]
A couple of comments/questions, but no blockers.

**Review by kentquirk — APPROVED** [review]

File: logsapi/server_test.go [hunk 2]
```diff
@@ -213,3 +212,13 @@ func TestTimestampsFunctionMessageWithDuration(t *testing.T) {
 	assert.Equal(t, ts.String(), event.Timestamp.String())
 
 }
+
+func TestLibhoneyEventWithSampleRate(t *testing.T) {
+	events := postMessages(t, []LogMessage{{
+		Time:   epochTimestamp,
+		Type:   "function",
+		Record: `{"time": "2020-12-25T12:34:56.789Z", "samplerate": 5, "data": {"foo": "bar", "duration_ms": 54} }`,
```
Comment (TylerHelmuth) [inline logsapi/server_test.go:220]: Can you add test cases for sample rate being a:
- float
- string
- an invalid type like boolean

File: CHANGELOG.md
Comment (TylerHelmuth) [inline CHANGELOG.md:1]: these changes can be removed, we'll update the changelog as part of the [release process](https://github.com/honeycombio/honeycomb-lambda-extension/blob/main/RELEASING.md)

**Review by TylerHelmuth — APPROVED** [review]

**Review by MikeGoldsmith — APPROVED** [review]

**Pull request merged**